### ISSUE TITLE
No interface is running right now

### ISSUE THREAD

**Issue by tushar-31093** [gradio-app/gradio#3060]
### Describe the bug

![image](https://user-images.githubusercontent.com/40886260/214615974-7c1809a3-c2d1-4cc7-9a86-c96510fd37b4.png)

Just host anything on EC2 and using the --share parameter the url crashes after a few hours despite the code running on the backend. The problem is consistent with all sorts of implementations.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

![image](https://user-images.githubusercontent.com/40886260/214615974-7c1809a3-c2d1-4cc7-9a86-c96510fd37b4.png)

Just host anything on EC2 and using the --share parameter the url crashes after a few hours despite the code running on the backend. The problem is consistent with all sorts of implementations.

### Screenshot

![image](https://user-images.githubusercontent.com/40886260/214615974-7c1809a3-c2d1-4cc7-9a86-c96510fd37b4.png)

### Logs

```shell
NA.. running it on a non-colab machine. I will run it again with this enabled to update it here.
```

### System Info

```shell
g5dn.4xlarge instance on AWS EC2
```

### Severity

blocking all usage of gradio

**Comment by freddyaboulton** [comment]
Hi @tushar-31093 !

To make sure I understand, the problem is that the link expires after a couple of hours (can you say how many)?

**Comment by tushar-31093** [comment]
I ran 2 instances yesterday around 10 p.m. 

I checked the links today morning and they are not running.

**Comment by MoonGotArts** [comment]
Been running into the same issue with the AUTOMATIC1111 UI for Stable Diffusion, links used to last the claimed 72 hours, but now they go bad at random times, sometimes as little as 1 hour.

**Comment by tushar-31093** [comment]
> Been running into the same issue with the AUTOMATIC1111 UI for Stable Diffusion, links used to last the claimed 72 hours, but now they go bad at random times, sometimes as little as 1 hour.

@freddyaboulton 

Yeah that too. I have Stable Diffusion, GPT-J and Neo running along with a few other prototypes, all use gradio and all are crashing on AWS EC2 instances.

Simultaneously, if i want to use the FastAPI version of loading it and then using it with NGNIX, it doesn't work then either, some config issue for sure. Cant tell.

**Comment by treksis** [comment]
same issue. using paperspace

**Comment by abidlabs** [comment]
Thanks all, we'll look into this next week.

**Comment by sgtdingo** [comment]
Same issue here running Automatic1111

**Comment by Weetile** [comment]
same issue here with automatic1111

**Comment by LumbarJack** [comment]
Same issue here on automatic 1111. Is there any useful info I could provide next time it happens?

**Comment by tushar-31093** [comment]
> Thanks all for creating this issue. It would be helpful for us to know 2 things:
> 
> * How long have you been experiencing this issue?
> * How long doe the interfaces typically run before crashing? Is it minutes, or a few hours? 

Started in the last week of December with some minor updates. Just performed a gir pull then. 

It crashed my codes at random times and in general the interface stopped showing after a day in the early days for both GPT-J, Automatic1111 stable diffusion, many similar implementations.

I assumed it was their code and let it go still some bug fixes were addressed.

Lately I built a custom dashboard for my team to use internally in early January. Basic crud operations with sqlite and S3 and a local DB too to test on the local machine.

It rendered perfectly but randomly crashes between 20 mins to 10 hours. 

Since then I re-visited all my previous codes, used --share and everything to try figure it out myself. But unfortunately I couldn't. 

This issue has been there for a while only that lately it's become worse. It's not stable and rendering stops despite the servers and the code still running in the background.

Hope this helps. Let me know if you need any futher details or I'm others others would share their views as well.

**Comment by abidlabs** [comment]
Thank you @tushar-31093, this is extremely helpful!

**Comment by aliabid94** [comment]
Huh will look into this!

**Comment by ThanapatSornsrivichai** [comment]
I have run into this problem 2 times in a row. 1 and 2 February.
Using latest Automatic1111 repo on TheLastBen Colab.
Both goes down after 4 hour-ish

**Comment by ThanapatSornsrivichai** [comment]
Got the same problem 3 days in a row.
Automatic1111 on colab with TheLastBen both at latest repo.
The colab still running but the link goes No Interface is running right now.
The link goes down after 3-4 hours on all three days.

**Comment by ramzillious** [comment]
Having the same problem trying to run automatic 1111 through google collab. As soon as it popped the link out it gave me that same problem. Been trying to get SD running in multiple ways on mac and havent had any luck. This time thought i was there, thought too soon

**Comment by abidlabs** [comment]
@ramzillious so to be clear, you had this problem *immediately* after the link was generated? Not a few hours after, like some of the previous users have reported?

**Comment by ramzillious** [comment]
@abidlabs yep immediately never got through to the SD page

**Comment by ramzillious** [comment]
This is the code. It is still running the process so Im not sure if its not done and I'm just being impatient. I just saw the link pop up and thought it was done. The link does look completely different than anything im seeing other people get though.
<img width="1179" alt="Screenshot 2023-02-06 at 1 16 38 PM" src="https://user-images.githubusercontent.com/122573313/217091103-28a4db5c-0584-4d7d-b23a-87f0c371791b.png">

**Comment by abidlabs** [comment]
That's frustrating, sorry it's been difficult bug for us to reproduce but we're looking into it!

**Comment by ramzillious** [comment]
@abidlabs definitely frustrating. Been trying different methods for 3 days anything from running it through terminal to google colab. Worth noting I am trying to run it on mac OS.

**Comment by ramzillious** [comment]
link to my reddit post trying to get help on this 
https://www.reddit.com/r/StableDiffusion/comments/10tjigh/unbelievably_hard_time_getting_stable_diffusion/

**Comment by ramzillious** [comment]
a little more in depth w screenshots

**Comment by ramzillious** [comment]
redid it in colab with SD 2.1 instead and it worked!

**Comment by abidlabs** [comment]
Oh that's very interesting... I don't know how that could have changed the gradio side of things, but good to know regardless. Thanks @ramzillious, and please let us know if the link goes down after a short while

**Comment by ramzillious** [comment]
thanks for the moral support haha ill let yall know if it does

**Comment by ramzillious** [comment]
welp literally took 30 minutes. link went down already

**Comment by ramzillious** [comment]
@abidlabs

**Comment by ramzillious** [comment]
Now I am getting this error at the end when i try to rerun the code in colab 
<img width="1635" alt="Screenshot 2023-02-06 at 4 51 56 PM" src="https://user-images.githubusercontent.com/122573313/217120320-545277de-cadc-4080-896b-8de332e31b1e.png">

**Comment by tushar-31093** [comment]
I would recommend that the team working on this, please don't bother about reproduction of the error. 

Let's try to have a stress test with 3 major applications across 3 versions of python.

Applications:
1. Stable Diffusion Web UI
2. Deploying GPT Neo 
3. Custom Application that does crud operations 

Run it using share.

Python versions: 
3.8, 3.9, 3.10

If you want I can do it for you and share the results. 

If you ask me something is off with the Gradio and fastapi mapping.

The localhost works perfectly fine. Even on AWS EC2, the instance where i deployed application 1 and 3. They are working fine with the exposed IPv4 http link.

Tested it with python 3.8, 3.10
Python venv not anaconda.

Issue is selectively with the share part and somewhat the bindings. 

It's hard for me to debug, but if you point me where to check for it I surely can. 

I tried binding Gradio with ngnix as well to serve my web app to my internal team but that failed miserably as well. Something off the ip and bindings. 

Maybe we can develop a workaround on this.

**Comment by abidlabs** [comment]
Thanks for the suggestions @tushar-31093, we're taking a deep dive ourselves, but will definitely reach out if we could use another hand.

> I tried binding Gradio with ngnix as well to serve my web app to my internal team but that failed miserably as well. Something off the ip and bindings.

This seems like a separate issue as this doesn't involve sharing, correct?

**Comment by abidlabs** [comment]
For those running the Stable Diffusion Web UI, I assume you are trying out the one by TheLastBen? https://colab.research.google.com/github/TheLastBen/fast-stable-diffusion/blob/main/fast_stable_diffusion_AUTOMATIC1111.ipynb#scrollTo=Y9EBc437WDOs

**Comment by abidlabs** [comment]
For those that have experienced this issue, is it possible that your computers disconnected from the network (e.g. computer went to sleep?)

**Comment by abidlabs** [comment]
Just FYI, our best hypothesis right now, is that Colab is disconnecting after a certain amount of "idle time". According to this [StackOverflow answer](https://stackoverflow.com/questions/57113226/how-can-i-prevent-google-colab-from-disconnecting?page=2&tab=scoredesc), the idle time that Colab allows is 90 minutes. 

If you think this explains or does not explain the behavior you're seeing, let us know!

**Comment by jmhessel** [comment]
+1 --- this is happening to me outside of a collab notebook. I am just running on a server, and the generated link dies sometimes as quickly as 30min, other times, it lasts maybe a few hours. The server is still running with `demo.launch(share=True)`.

**Comment by abidlabs** [comment]
That's very helpful @jmhessel. We'll do some more digging.

**Comment by jmhessel** [comment]
it's a local server where I work similar to a workstation. I am happy to try hosting at the link and then DM it if you want to monitor... it could be the case that the connection dropped, but I have been sshed into it and haven't noticed that

**Comment by abidlabs** [comment]
I think the server details are enough information to go off of right now. Will reach out if we need more info, thanks again @jmhessel

**Comment by tushar-31093** [comment]
Hey,
It's the original automatic1111 repo on AWS EC2 instance not colab notebooks.

The issue is still there. Just hosted a telegram bot on AWS that uses openai's API structure to answer questions and generate content. That too stopped working after 2 hours.

**Comment by sgtdingo** [comment]
I concur -- the issue is still very much present in the Automatic1111 repo
when run locally and not through Colab.

**Comment by Caneloide** [comment]
I'm having this same issue but while running Auto1111 on Colab. I was trying to train an embedding; 10k steps, every 500 a copy was saved. But when it was at 499 steps, the interface crashed. I tried it 4 or 5 times more, changing some parameters, with less steps, reloading it, but nothing worked. The same error popped out no matter how long the interface has been running on the Colab notebook. Sometimes it just stopped at 499 steps, sometimes at 345. The error "no interface is running right now" continued popping out. I'm trying to do this on a Macbook (not a new one, btw) but I'm using the hardware resources of the Colab. Even in this way, it didn't work. 
The strange thing here is that the code in the Colab notebook continued running and training 'till it reaches the 500 steps to save the first embedding. When it comes to the 500 step, it crashes and stop running. I have been trying to do this since yesterday at night. No matter the hour, the parameters or the task, it just crashes spontaneously.

**Comment by AdamHardix** [comment]
I've been having this issue since late December/early January with Automatic1111's Stable Diffusion WebUI, and I run it locally, so it couldn't be an issue with Google Colab. I have two instances that I run on my home PC, and am able to access them from any device on my home network as I have the --listen argument enabled. However, if I use --share, the Gradio link fails fairly consistently at around 30 minutes, despite being able to access both instances on the localhost or through the home network just fine.

Could this be an issue from Gradio having been updated around the time this issue started? And, if so, is there a way to roll back to a previous version that wasn't almost unusable with Automatic1111's Stable Diffusion WebUI?

**Comment by ramzillious** [comment]
Anyone know anything other than gradio and automatic 1111 that we can use for stable diffusion? Seems like this has been a pretty consistent problem with gradio and auto 1111 and I want to see if I can use any other versions of stable diffusion on my macbook until it gets fixed. I am running an M1 with 64gb of ram and all specs maxed out cause I use it for multiple forms of animation so it definitely isn't a hardware issue.

**Comment by abidlabs** [comment]
@ramzillious and anyone else trying to use the automatic1111 GUI, you could check the version hosted on Hugging Face Spaces: https://huggingface.co/spaces/camenduru/webui

(Since it's hosted and running locally, you won't have to worry about the share timing out!)

**Comment by ThanapatSornsrivichai** [comment]
Any updates on the issue? I still got the issue at 90% rate. Crashing within 3-4 hours. Using TheLastBen fast stablediffusion on google colab. With 4-5 users using the interface at the same time. Out of 10 days, only 1 day that I guess got lucky and able to run for 24 hours.

**Comment by abidlabs** [comment]
We're working on a fix, hoping to release this week!

**Comment by AdamHardix** [comment]
> Anyone know anything other than gradio and automatic 1111 that we can use for stable diffusion? Seems like this has been a pretty consistent problem with gradio and auto 1111 and I want to see if I can use any other versions of stable diffusion on my macbook until it gets fixed. I am running an M1 with 64gb of ram and all specs maxed out cause I use it for multiple forms of animation so it definitely isn't a hardware issue.

There's InvokeAI, though I don't know if it has a remote access option like Automatic1111's webUI. I've installed it, but have yet to mess with it.

**Comment by abidlabs** [comment]
Hi folks! We believe we've found a fix. We've released version `gradio==3.18.1b2` for easier testing. It would be super helpful if you could try this out and let us know if this solves the issue before we do a wider release. To test, do the following steps:

1. `pip install gradio==3.18.1b2` 
2. Create a gradio demo and launch it with `share=True`
3. Leave it running for many hours and ensure that that the share link continues to work

Thanks!

**Comment by tushar-31093** [comment]
> Hi folks! We believe we've found a fix. We've released version `gradio==3.18.1b2` for easier testing. It would be super helpful if you could try this out and let us know if this solves the issue before we do a wider release. To test, do the following steps:
> 
> 1. `pip install gradio==3.18.1b2` 
> 2. Create a gradio demo and launch it with `share=True`
> 3. Leave it running for many hours and ensure that that the share link continues to work
> 
> Thanks!
> 

Perfect. On it.

**Comment by jwkirchenbauer** [comment]
Also hopping on this, will report if stable!

**Comment by Caneloide** [comment]
> Any updates on the issue? I still got the issue at 90% rate. Crashing within 3-4 hours. Using TheLastBen fast stablediffusion on google colab. With 4-5 users using the interface at the same time. Out of 10 days, only 1 day that I guess got lucky and able to run for 24 hours.

Does this also work for the Auto1111 on Colab?

**Comment by robinjhuang** [comment]
For some reason after I upgrade to this version (3.18.1b2) when I run a demo with Share=True, the public gradio link does not show in the output. Only the local link.

**Comment by Caneloide** [comment]
> Hi folks! We believe we've found a fix. We've released version `gradio==3.18.1b2` for easier testing. It would be super helpful if you could try this out and let us know if this solves the issue before we do a wider release. To test, do the following steps:
> 
> 1. `pip install gradio==3.18.1b2`
> 2. Create a gradio demo and launch it with `share=True`
> 3. Leave it running for many hours and ensure that that the share link continues to work
> 
> Thanks!

Does this also work for the Auto1111 on Colab?

**Comment by Caneloide** [comment]
I still got the issue when I try to train an embedding on the LastBen Colab. Crashing within some minutes after reaching step 499; I guess when trying to save the first log directory. It's been happening to me for weeks and it's still happening today when I tried again. I am running v2-768. Also, in the past days I tried to use SD without training anything, just with the model, and it crashes while generating images after 35-40 minutes or even less. The SS i'm attaching is from today while training the embedding. 
![Captura de Pantalla 2023-02-17 a la(s) 1 19 11](https://user-images.githubusercontent.com/124810249/219581098-a0227fd6-38be-42f2-8c01-67e4c07dc543.png)

**Comment by jwkirchenbauer** [comment]
> For some reason after I upgrade to this version (3.18.1b2) when I run a demo with Share=True, the public gradio link does not show in the output. Only the local link.

I have the same issue

**Comment by tushar-31093** [comment]
And I cannot event install it :(

![image](https://user-images.githubusercontent.com/40886260/219588461-5de4b1bf-e8a5-456f-8e80-4c789188d4f0.png)

**Comment by abidlabs** [comment]
Thanks everyone for trying this out! 

@robinjhuang @jwkirchenbauer I think the issue is that since you upgraded from an older version of gradio, it's using the older version of the sharing file that's still on your computer. You can see that on a [fresh installation in Google Colab, the share link appears](https://colab.research.google.com/drive/1RoQAqhBDhX2oXiqlIZS6ZE9BNLe9ZRTK#scrollTo=mQQhH02tMpKC). I'm going to test this hypothesis and release a new version with the fix, thanks for the feedback! 

@Caneloide for the Automatic1111 web ui, the maintainers of the web ui will need to update the version of gradio used in the web ui *after* we release a stable version to pypi. For now, I would suggest testing with your own gradio demos

@tushar-31093 that sounds like a network issue that is preventing you from connecting to pypi?

**Comment by tushar-31093** [comment]
> Thanks everyone for trying this out!
> 
> @robinjhuang @jwkirchenbauer I think the issue is that since you upgraded from an older version of gradio, it's using the older version of the sharing binary file that's still on your computer. You can see that on a [fresh installation in Google Colab, the share link appears](https://colab.research.google.com/drive/1RoQAqhBDhX2oXiqlIZS6ZE9BNLe9ZRTK#scrollTo=mQQhH02tMpKC). I'm going to test this hypothesis and release a new version with the fix, thanks for the feedback!
> 
> @Caneloide for the Automatic1111 web ui, the maintainers of the web ui will need to update the version of gradio used in the web ui _after_ we release a stable version to pypi. For now, I would suggest testing with your own gradio demos
> 
> @tushar-31093 that sounds like a network issue that is preventing you from connecting to pypi?

Not really. Going to try a fresh installation again on my EC2 server. Let's see. Will get back to you here in a bit.

**Comment by abidlabs** [comment]
Just released `gradio==3.18.1b7` with the fix for the issue that @robinjhuang and @jwkirchenbauer were experiencing. Could you please give this a try and let me know if it works for you? 

Same testing instructions as above:

1. pip install `gradio==3.18.1b7`
2. Create a gradio demo and launch it with `share=True`
3. Leave it running for many hours and ensure that that the share link continues to work

**Comment by tushar-31093** [comment]
> > Hi folks! We believe we've found a fix. We've released version `gradio==3.18.1b2` for easier testing. It would be super helpful if you could try this out and let us know if this solves the issue before we do a wider release. To test, do the following steps:
> > 
> > 1. `pip install gradio==3.18.1b2`
> > 2. Create a gradio demo and launch it with `share=True`
> > 3. Leave it running for many hours and ensure that that the share link continues to work
> > 
> > Thanks!
> 
> Does this also work for the Auto1111 on Colab? How can I add that code into Colab? In which cell can I put it or where do I have to enter it?

Nope. Stopped again. It was deployed on AWS EC2

![image](https://user-images.githubusercontent.com/40886260/219846524-d6d0908c-dc41-4d7f-bd86-34b58ef90c61.png)

**Comment by abidlabs** [comment]
@tushar-31093 from the format of the share link, it looks like it's using the older version of gradio. The older version included two hypens, and the new version does not. Are you sure you're running `gradio==3.18.1b7`?

**Comment by tushar-31093** [comment]
> @tushar-31093 from the format of the share link, it looks like it's using the older version of gradio. The older version included two hypens, and the new version does not. Are you sure you're running `gradio==3.18.1b7`?

Yes. It was a fresh installation after removing the previous one. Removed it from the cache too.

**Comment by abidlabs** [comment]
What do you see when you run the following? 

```py
import gradio as gr
print(gr.__version__)
```

**Comment by tushar-31093** [comment]
> What do you see when you run the following?
> 
> ```python
> import gradio as gr
> print(gr.__version__)
> ```

![image](https://user-images.githubusercontent.com/40886260/219847940-93e45dc1-e4f1-4982-ad23-e1f11a34f70e.png)

**Comment by KaiWangCanada** [comment]
Just got the same issue. Restart solved it.

**Comment by tushar-31093** [comment]
> > What do you see when you run the following?
> > 
> > ```python
> > import gradio as gr
> > print(gr.__version__)
> > ```
> 
> ![image](https://user-images.githubusercontent.com/40886260/219847940-93e45dc1-e4f1-4982-ad23-e1f11a34f70e.png)
> 
> I can try running it again with a fresh installtion of gradio as well if you'd like.

I have launched it again. The format is as per what you suggested. Let's see if it stands. Will let you know by tomorrow if it crashes or works as is.

One thing I'd like to point out is, the whole UI/UX is gone with this update. Sizes, dimensions etc... It's acting up on all different implementations I've tried. 

- Stable Diffusion (Automatic1111)
- Gradio Crud
- Hugging face implementation of a Chatbot using GPT3 API.

**Comment by ThanapatSornsrivichai** [comment]
Could confirmed with @tushar-31093 all the UI/UX is messed up and not working on Window/Mac/Mobile when using 3.18.1b7 version.
Automatic1111 webui run on local.

**Comment by abidlabs** [comment]
@tushar-31093 @ThanapatSornsrivichai the Automatic1111 UI contains large amounts of custom CSS code, which we cannot guarantee work across Gradio versions. The maintainers of the UI are aware of this and change the CSS code between versions. We *do* aim for complete backwards compatibility from the Python API side, so if your app does not have custom CSS, it should not break if you upgrade to `3.18.1b7`.

If you are experiencing an issue with the UI/UX, feel free to share your Python code here and I'm happy to take a look. 

Did you experience any issues with the share link timing out after you've upgraded to `3.18.1b7`?

**Comment by ThanapatSornsrivichai** [comment]
@abidlabs  I'm currently testing with the share link timing on 3.18.1b7 so far so good 6hours passed.
Below was the custom the css from Automatic1111. It works with 3.16.2 version.
https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/style.css

I'll update with the result of share link again. Thank you for looking into the problem.

**Comment by tushar-31093** [comment]
> @tushar-31093 @ThanapatSornsrivichai the Automatic1111 UI contains large amounts of custom CSS code, which we cannot guarantee work across Gradio versions. The maintainers of the UI are aware of this and change the CSS code between versions. We _do_ aim for complete backwards compatibility from the Python API side, so if your app does not have custom CSS, it should not break if you upgrade to `3.18.1b7`.
> 
> If you are experiencing an issue with the UI/UX with a Gradio app that does _not_ include custom CSS, feel free to share your Python code here and I'm happy to take a look.
> 
> Did you experience any issues with the share link timing out after you've upgraded to `3.18.1b7`?

The share link is working. That's perfect. Thank you all for working it through. For the specific use-cases, I get your point about the customizations they might have done. Once the gradio version is updated, they will have to fix it anyway. No issues.

Other than this discussion point, I had 1 thing to ask and that is hosting it on my own EC2 server. Right now I am using your toolkit to provide me with an SSL enabled share link for 72 hours. What if I want to bind the local host to NGNIX and deploy it on the server I have? I tried doing it with the local host setting but I think it does not get detected by NGNIX or the settings need to be something else. Do you have any suggestions here or if there is a way to use gradio + nginix configuration to host a web app alltogether.

**Comment by ThanapatSornsrivichai** [comment]
> @tushar-31093 @ThanapatSornsrivichai the Automatic1111 UI contains large amounts of custom CSS code, which we cannot guarantee work across Gradio versions. The maintainers of the UI are aware of this and change the CSS code between versions. We _do_ aim for complete backwards compatibility from the Python API side, so if your app does not have custom CSS, it should not break if you upgrade to `3.18.1b7`.
> 
> If you are experiencing an issue with the UI/UX with a Gradio app that does _not_ include custom CSS, feel free to share your Python code here and I'm happy to take a look.
> 
> Did you experience any issues with the share link timing out after you've upgraded to `3.18.1b7`?

Is it possible to do this fix on gradio 3.16.2? since it was used for Automatic1111 and I don't think they will update to 3.18.1b7 anytime soon.

**Comment by abidlabs** [comment]
> The share link is working. That's perfect

That's good to hear, thanks for testing!

> Other than this discussion point, I had 1 thing to ask and that is hosting it on my own EC2 server. Right now I am using your toolkit to provide me with an SSL enabled share link for 72 hours. What if I want to bind the local host to NGNIX and deploy it on the server I have?

This is definitely possible. We have a number of folks that have done this. You might want to take a look at this issue: https://github.com/gradio-app/gradio/issues/2979. If you have further questions, feel free to ask in GitHub Discussions as this is not really a bug fix or a feature request

**Comment by abidlabs** [comment]
> Is it possible to do this fix on gradio 3.16.2?

We can't retroactively change `gradio 3.16.2` but there might be a workaround that lets you do this. The following assumes you are running in a Colab environment:

1. First run Automatic1111 web ui (or any gradio app) **locally** (without `share=True`)
2. Then download the v2 share binary by running the following:

```py
import requests
import os
import stat

binary_url = f"https://cdn-media.huggingface.co/frpc-gradio-0.2/frpc_linux_amd64"
resp = requests.get(binary_url)

if resp.status_code == 403:
    raise OSError(
        f"Cannot set up a share link as this platform is incompatible. Please "
        f"create a GitHub issue with information about your platform: {platform.uname()}"
    )

resp.raise_for_status()

binary_path = "frpc_linux_amd6"
# Save file data to local copy
with open(binary_path, "wb") as file:
    file.write(resp.content)
st = os.stat(binary_path)
os.chmod(binary_path, st.st_mode | stat.S_IEXEC)
```

3. Then run the following terminal command to generate the share link:

```bash
!/content/frpc_linux_amd64 http -n $RANDOM -l 7860 -i 127.0.0.1 --uc --sd random --ue --server_addr 35.165.217.216:7000 --disable_log_color
```

Although I tested this and it seemed to work, I should emphasize that this is NOT officially supported and may not work reliably.

**Comment by ThanapatSornsrivichai** [comment]
> 

How do I run this locally? I've got the frpc_linux_amd64 but I couldn't run
 frpc_linux_amd64 http -n $RANDOM -l 7860 -i 127.0.0.1 --uc --sd random --ue --server_addr 35.165.217.216:7000 --disable_log_color
It doesn't generate a share link.

Is there a way to force gradio 3.16.2 to use frpc-gradio-0.2? as I saw in the issue, it seems that the problem occurs when it try to use frpc-gradio-0.1 instead.

**Comment by abidlabs** [comment]
You have to delete `frpc_linux_amd64` if you downloaded it previously and instead download the 0.2 version. That's what the Python code I shared above does. Then you have to create the share link above by running the terminal command above.

Here's a Colab notebook showing the process step-by-step (I simplified the code a little bit): https://colab.research.google.com/drive/13r0tM7HLzbnEvRdmuoY3eiLj2W_Im0qf#scrollTo=WIVd09klM_BE

Like I said before though, this isn't an official solution and we won't be able to invest effort into ensuring it works.

**Comment by ThanapatSornsrivichai** [comment]
> You have to delete `frpc_linux_amd64` if you downloaded it previously and instead download the 0.2 version. That's what the Python code I shared above does. Then you have to create the share link above by running the terminal command above.
> 
> Here's a Colab notebook showing the process step-by-step (I simplified the code a little bit): https://colab.research.google.com/drive/13r0tM7HLzbnEvRdmuoY3eiLj2W_Im0qf#scrollTo=WIVd09klM_BE
> 
> Like I said before though, this isn't an official solution and we won't be able to invest effort into ensuring it continues to work reliably.

Sorry for the question. I've successfully download the  `frpc_linux_amd64` with the provided code. But I couldn't run the 

!/content/frpc_linux_amd64 http -n $RANDOM -l 7860 -i 127.0.0.1 --uc --sd random --ue --server_addr 35.165.217.216:7000 --disable_log_color

equivalent on locally window10. Currently I'm not using colab anymore.

**Comment by ThanapatSornsrivichai** [comment]
I tried connect colab to local runtime and still couldn't execute

 frpc_linux_amd64 http -n $RANDOM -l 7860 -i 127.0.0.1 --uc --sd random --ue --server_addr 35.165.217.216:7000 --disable_log_color
 
![image](https://user-images.githubusercontent.com/36567807/220030502-16f52f0f-b153-4aa8-9344-3db6a10df64c.png)

Is it linux only?

**Comment by ThanapatSornsrivichai** [comment]
I have installed ubuntu instead of using window. Now the workaround for 3.16.2 is working. I'll update on the result if the shared link stays on. Thank you very much.

**Comment by abidlabs** [comment]
Great! Now that we've released `gradio==3.19`, I'm going to cautiously close this issue. Hopefully this release resolves sharing for everyone!

**Comment by ProjectRevoTPP** [comment]
I installed gradio 3.19 for my stable diffusion setup, but this still keeps happening. Dont think this is fixed.

**Comment by abidlabs** [comment]
Hi @ProjectRevoTPP are you using the Automatic1111 Web UI? The webui uses a pinned version of Gradio so it may not actually be using Gradio 3.19.1 under the hood

**Comment by tushar-31093** [comment]
> Hi @ProjectRevoTPP are you using the Automatic1111 Web UI? The webui uses a pinned version of Gradio so it may not actually be using Gradio 3.19.1 under the hood

You are right. It needs to be changed in their requirements file. I did that only once you suggested that they pinned the version. That is what made their UI off and go everywhere 🤣

**Comment by ProjectRevoTPP** [comment]
What I did was run venv/Scripts/Activate.ps1 and manually did python -m pip install gradio==3.19 to ensure my SD install is using that version. It is using it, it still isnt working.

**Comment by jmhessel** [comment]
I successfully ran 2 interfaces for 72h with `3.18.1b7` . thanks for the change --- it worked for me for these cases.

**Comment by ThanapatSornsrivichai** [comment]
> > Hi @ProjectRevoTPP are you using the Automatic1111 Web UI? The webui uses a pinned version of Gradio so it may not actually be using Gradio 3.19.1 under the hood
> 
> You are right. It needs to be changed in their requirements file. I did that only once you suggested that they pinned the version. That is what made their UI off and go everywhere 🤣

3.19.1 Solved gradio the issue but now broke everything else on Automatic1111

**Comment by ProjectRevoTPP** [comment]
I can confirm it's fixed this time, the gradio link didnt die while I was at work.

**Comment by Hilurex** [comment]
hello, I have the same problem, I don't know whether to update since they say that they are giving you problems in automatic 1111 and the interface stops working or gives errors

**Comment by Zephalon** [comment]
This is not yet resolved. I started it before my work at around 7:30 and around 12:30 it broke. When I came home it worked fine locally.

Latest version with Gradio 3.23.

**Comment by abidlabs** [comment]
@Hilurex has it continued to happen or was it a one-time thing?

**Comment by Hilurex** [comment]
@abidlabs not so much, but now is when the system hangs at the time of creating an image, when I create an image it does not show and / or does not show the progress in the command console

**Comment by akatsukiaashish** [comment]
Hi all! I'm using gradio version 1.32 and still encountering this issue. Gradio link expires roughly after 30 mins to 1 hour but backend code keeps working fine on local machine. I am 
![Screenshot_20230609_163152_Chrome](https://github.com/gradio-app/gradio/assets/35386435/4e75f3cf-8343-42d8-83d1-4976ec89869c)
using stable diffusion webui on windows 11.

**Comment by abidlabs** [comment]
What version of Gradio are you using? I'm pretty sure `1.32` is a typo

**Comment by timmbobb** [comment]
I am having this bug still. Running the latest version of stable-diffusion webui on my pc and using --share with authentication and it will run fine for 30 minutes - 2 hours but then I get the infamous "No interface is running right now" message. My console on my pc looks completely normal but it's definitely no longer able to be connected to. Tried a brand new install of stable diffusion, no dice.

**Comment by akatsukiaashish** [comment]
You're right. Apologies. I meant to say 3.32.

**Comment by AI-Robot-Morris** [comment]
same here 
use webui a1111 with gradio ver-3.32.0 --share service at 5am, but encounter "No interface is running right now" at 7pm.
Try to use ver 3.33.1 and see how its going.

**Comment by AdamHardix** [comment]
Seven months later, this is _still_ a problem. And it's not even consistent. Sometimes, it might last a few hours: others, five minutes or less. The --share argument should be entirely removed from the app. It's worthless.

**Comment by timmbobb** [comment]
I've got a bit of a solution. I've been running an ngrok tunnel which basically just gives you a link which redirects to localhost:3000 on my machine. It's got the same type of authentication where you can set up a username and password.

**Comment by jgemeigh** [comment]
Here to report that this is still happening at the end of July 2023, and has been reported since the end of January 2022 or late December even.

The link with gradio used to last 72 hours but now only lasts between 1 minute and a few hours, which is impossible to test since the interface doesn't say "this happened" in the log.

The logical suggestion to me, would be that Gradio is not honoring their 72 hour lease (let's be real, at best it only ever lasted a little over 2 days, never the full 3 even when this was brand new stuff and there were fewer links being leased.

Can somebody get in touch with a gradio expert to like, run that one past them? 

This is a silly thing to be happening across multiple versions and months and fresh installs.

**Comment by RevoSucks** [comment]
Yeah, this is still happening as of `3.32.0`.

**Comment by ilayasayag** [comment]
same error here for a new user.
what is the best way to approach this?

**Comment by freddyaboulton** [comment]
Hi @ilayasayag - sorry about that. The share links are working now for me

<img width="638" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/ab9fb634-cdce-46b1-924a-a0b904d77475">

**Comment by timmbobb** [comment]
> same error here for a new user. what is the best way to approach this?

Look up Ngrok. You can set up an ngrok tunnel with a password and username which allows you to connect to the localhost:3000 through a link it gives you when you run it. You have to make an account but it's definitely worth it. Will run forever if you don't shut it off. You don't have to have the -listen argument anymore either with this.

**Comment by ilayasayag** [comment]
> Hi @ilayasayag - sorry about that. The share links are working now for me
> 
> <img alt="image" width="638" src="https://user-images.githubusercontent.com/41651716/262052944-ab9fb634-cdce-46b1-924a-a0b904d77475.png">

Hi, still no change. It worked yesterday but today whenever I try to reload new model it crashes

Any suggestions?

**Comment by maphim** [comment]
Replace !sed -i 's/gradio==3.32.0/gradio==3.18.1b7/g' requirements.txt
!COMMANDLINE_ARGS="--share --gradio-debug" REQS_FILE="requirements.txt" python launch.py

**Comment by HamzaSidhu** [comment]
I am currently using Gradio version 3.41.2 and encountering issues. Could someone please advise me on which version is currently stable and functioning properly? Additionally, I would like to prevent links from expiring after 72 hours in the Automatic1111 Webui.

**Comment by thiswillbeyourgithub** [comment]
In my case this happens only if I use SSL (self signed, so with ssl_verify ot True on demo.launch()). If I use share I disable SSL and it works.

I don't think this should happen.

Link #5622

**Comment by SURE945** [comment]
> In my case this happens only if I use SSL (self signed, so with ssl_verify ot True on demo.launch()). If I use share I disable SSL and it works.
> 
> I don't think this should happen.
> 
> Link #5622

I meet the same problem in ChatRTX 0.3.

**Comment by cageyoko** [comment]
> In my case this happens only if I use SSL (self signed, so with ssl_verify ot True on demo.launch()). If I use share I disable SSL and it works.
> 
> I don't think this should happen.
> 
> Link #5622

I met same problem today, have you found some method to solve it ?

**Comment by Tarrachal** [comment]
I am having this issue as well. It's happened every time I've run it and it crashes within 2-3 hours. Everything else is great.

**Comment by ZRabinia** [comment]
This is because of your session crashed from too many requests. Try to run just the related cells and it will work.

**Comment by Lord451125** [comment]
> Hi guys , try setting live=True in the end of the interface ,it will automatically refresh the interface

Tried this, still died around 3 and a half hours mark.  Assuming I did it right.  Went to the directory/file: venv\Lib\site-packages\gradio\interface.py  and changed "live: bool = False," to "live: bool = True,".  Not sure if I needed to do something else.
